asmcomp/amd64/emit.mlp

```diff
@@ -615,14 +615,16 @@ let emit_instr fallthrough i =
         end;
         cfi_remember_state ();
         I.mov (domain_field Domainstate.Domain_c_stack) rsp;
-        (* FIXME: CFI *)
-        (* cfi_def_cfa_offset 8; *)
+        cfi_def_cfa_offset 0;
+
+        (* Keep GDB happy. If the cfa_offset was left at 0, then GDB complains
+         * that the frame did not save the PC and truncates the backtrace.
+         * Subtract rsp by 16 (instead of 8) to keep the stack aligned to
+         * 16-byte boundary. *)
+        I.sub (int 16) rsp;
+        cfi_adjust_cfa_offset 16;
+
         emit_call func;
-        (* FIXME KC:
-        if Config.spacetime then begin
-          record_frame Reg.Set.empty false i.dbg ~label:label_after
-        end
-        *)
         I.mov rbp rsp;
```

Comment:
We should free the stack we reserved, 

```suggestion
I.add (int 16) rsp;
cfi_adjust_cfa_offset -16;
I.mov rbp rsp;
```

Reply:
We don't need to since the next time we make a c call, we will use the `Domainstate.Domain_c_stack` value, which has not been updated.

Reply:
Oh yeah. I agree, we allocated that space on the new C stack frame, which we are anyways throwing away.